aws-datastore/src/main/java/com/amplifyframework/datastore/DataStoreConfiguration.java

```diff
@@ -0,0 +1,387 @@
+    private final DataStoreConflictHandler dataStoreConflictHandler;
+    private final Integer syncMaxRecords;
+    private final Integer syncPageSize;
+    private Long syncIntervalInMinutes;
+    private Long syncIntervalMs;
+
+    private DataStoreConfiguration(
+            DataStoreErrorHandler dataStoreErrorHandler,
+            DataStoreConflictHandler dataStoreConflictHandler,
+            Long syncIntervalInMinutes,
+            Integer syncMaxRecords,
+            Integer syncPageSize) {
+        this.dataStoreErrorHandler = dataStoreErrorHandler;
+        this.dataStoreConflictHandler = dataStoreConflictHandler;
+        this.syncMaxRecords = syncMaxRecords;
+        this.syncPageSize = syncPageSize;
+        if (syncIntervalInMinutes != null) {
+            this.syncIntervalInMinutes = syncIntervalInMinutes;
+            this.syncIntervalMs = TimeUnit.MINUTES.toMillis(syncIntervalInMinutes);
+        }
+    }
+
+    /**
+     * Begin building a new instance of {@link DataStoreConfiguration}.
+     * @return A new builder instance
+     */
+    @NonNull
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Begin building a new instance of {@link DataStoreConfiguration} by reading DataStore
+     * settings from the config file and an optional set of user-provided overrides.
+     * @param pluginJson DataStore plugin configuration from amplicationconfiguration.json
+     * @param userProvidedConfiguration An instance of {@DataStoreConfiguration} with settings specified by the user
+     *                                  which will be used as overrides.
+     * @return A new builder instance
+     * @throws DataStoreException exception thrown if there's an unexpected configuration key or
+     * an invalid configuration value
+     */
+    @NonNull
+    static Builder builder(@NonNull JSONObject pluginJson,
+                           @Nullable DataStoreConfiguration userProvidedConfiguration) {
+        return new Builder(pluginJson, userProvidedConfiguration);
+    }
+
+    /**
+     * Begin building a new instance of {@link DataStoreConfiguration}.
+     * @param pluginJson The dataStore configuration as a JSONObject
+     * @return A new builder instance
+     * @throws DataStoreException exception thrown if there's an unexpected configuration key or
+     * an invalid configuration value
+     */
+    @NonNull
+    static Builder builder(@NonNull JSONObject pluginJson) {
+        return builder(pluginJson, null);
+    }
+
+    /**
+     * Creates an {@link DataStoreConfiguration} which uses all default values.
+     * @return A default {@link DataStoreConfiguration}
+     * @throws DataStoreException exception thrown if there's an unexpected configuration key or
+     * an invalid configuration value
+     */
+    @NonNull
+    public static DataStoreConfiguration defaults() throws DataStoreException {
+        return builder()
```

Comment:
Should this also include our opinionated error and conflict handlers?

Reply:
Those get set in the default constructor of the builder since there's no ambiguity with those two fields. That being said, it doesn't hurt to be explicit about it and setting it here.